pkg/web/helpers.go

```diff
@@ -0,0 +1,43 @@
+package web
```

Comment:
in the end users don't know if the file is context_keys or helpers or util. It's used as `web.<function>`. I can argue that operations should be separated semantically and `UserFromContext` and `NewContextWithUser` should be in a package whose purpose is user management. `pkg/web` doesn't need to know about users and correlation ids in my opinion. 
On the other hand k8s apiserver has a package `request` that has both context and request operations which would be similar to what you mean - https://github.com/kubernetes/apiserver/tree/master/pkg/endpoints/request 

I'll leave context_keys as is with the remark that it would be best if the functions were moved to a user-centric package.
The `CorrelationIDForRequest` is moved to `pkg/log/request.go`